src/components/ButtonComponent/ButtonComponent.module.css

```diff
@@ -0,0 +1,62 @@
+/* Base style for all buttons */
+.button {
+  font-family: 'Roboto', sans-serif;
+  padding: 10px 20px;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: background-color 0.3s ease;
+}
+
+/* Variant styles */
+.Compact {
+  padding: 5px 10px;
+  font-size: 12px;
+}
+
+.Regular {
+  padding: 10px 20px;
+  font-size: 14px;
+}
+
+.Large {
+  padding: 15px 30px;
+  font-size: 16px;
+}
+
+/* Color styles using enum names */
+.Blue {
+  background-color: blue;
+  color: white;
+}
+
+.Yellow {
+  background-color: yellow;
+  color: black;
+}
+
+.Green {
+  background-color: green;
+  color: white;
+}
+
+.Grey {
+  background-color: grey;
+  color: white;
+}
+
+.Red {
+  background-color: red;
+  color: white;
+}
+
+.Black {
+  background-color: black;
+  color: white;
+}
+
+.button:disabled,
+.button.loading {
+  background-color: #cccccc;
```

Comment:
Let's not gray out the button while it is loading. Let's just show the spinner. Can we give the button another property called `disabled` and when that is `true` we can gray out the button?